src/main/java/com/novauc/MessageController.java

```diff
@@ -0,0 +1,47 @@
+package com.novauc;
+
+import com.sun.javafx.sg.prism.NGShape;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.servlet.http.HttpSession;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by jerieshasmith on 3/15/17.
+ */
+@Controller
+public class MessageController {
+
+    @Autowired
+    MessageRepository messages;
+
+    @RequestMapping(path = "/", method = RequestMethod.GET)
+    public String home(Model model) {
+        List<Message> messageList = (ArrayList) messages.findAll();
+        model.addAttribute("messages", messageList);
+        return "home";
+
+
+    }
+
+    @RequestMapping(path = "/add-message", method = RequestMethod.POST)
+    public String addMessage(HttpSession session, String text) {
+        Message message = new Message(text);
+        messages.save(message);
+        return "redirect:/";
+    }
+        @RequestMapping(path ="/edit-message", method =RequestMethod.POST)
+        public String editMessage(Integer id,String text){
+       int l = Integer.valueOf(id);
+       Message m = messages.findOne(l);
+       m.setText(text);
```

Comment:
Good job Olympew.  You don't need line 40.  You should be able to just use id it comes in as an Integer.